vector/src/main/java/im/vector/app/features/widgets/webview/WebviewPermissionUtils.kt

```diff
@@ -15,17 +15,30 @@
  */
 package im.vector.app.features.widgets.webview
 
+import android.Manifest
 import android.annotation.SuppressLint
 import android.content.Context
 import android.webkit.PermissionRequest
+import androidx.activity.result.ActivityResultLauncher
 import androidx.annotation.StringRes
+import androidx.fragment.app.FragmentActivity
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import im.vector.app.R
+import im.vector.app.core.utils.checkPermissions
 
 object WebviewPermissionUtils {
 
+    private var permissionRequest: PermissionRequest? = null
```

Comment:
I'm a little weary of including mutable static state in the `object` instance, would prefer for this class to become instantiable (and then injected where needed) or have the activity/fragment to hold on to the state instead

Reply:
I had also thought of this, yeah. I had previously decided against putting the state into the fragment because that way state and logic are split between the fragment and `WebviewPermissionUtils` which I found harder to reason about. I made the utils into a normal class now and used a private instance variable on the fragment. If this needs to be injectable, I might need some help with that. 🙈